etc/cmake/options.cmake

```diff
@@ -44,6 +44,22 @@ endif()
 option(OT_BACKBONE_ROUTER "enable backbone router functionality")
 if(OT_BACKBONE_ROUTER)
     target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE=1")
+    set(OT_BACKBONE_ROUTER_DUA_NDPROXYING ON CACHE BOOL "Enable DUA NDProxying by default")
```

Comment:
Is it necessary?
`OPENTHREAD_CONFIG_BACKBONE_ROUTER_DUA_NDPROXYING_ENABLE` will take the value of `OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE` if not explicitly configured in the configuration

https://github.com/openthread/openthread/blob/524d8fbd6353b35f406ab199aaad011541b96760/src/core/config/backbone_router.h#L48-L57

Reply:
If we don't set `OT_BACKBONE_ROUTER_DUA_NDPROXYING` ON in cache, the later option definition will override it to off.

```cmake
option(OT_BACKBONE_ROUTER_DUA_NDPROXYING "enable Backbone Router DUA ND Proxying functionality" OFF)
if(OT_BACKBONE_ROUTER_DUA_NDPROXYING)
    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_BACKBONE_ROUTER_DUA_NDPROXYING_ENABLE=1")
else()
    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_BACKBONE_ROUTER_DUA_NDPROXYING_ENABLE=0")
endif()
```